### PR TITLE
Express material names in the 3DOM scene graph

### LINKED ISSUE #1062: 3DOM material names are not being expressed
When accessing materials in a 3DOM scene graph worker, names are not readable in the public API for `Material`. Names need to be expressed in the public API for `Material`.

### PR DESCRIPTION
**Pull request by cdata**
Fixes #1062

### CODE REVIEW

File: packages/3dom/src/context.ts [hunk 1]
```diff
@@ -161,6 +161,7 @@ export class ThreeDOMExecutionContext extends EventTarget {
     const port = await this[$workerInitializes];
     const {port1, port2} = new MessageChannel();
 
+    console.log(modelGraft?.model.toJSON());
```
Comment (elalish) [inline packages/3dom/src/context.ts]: Sure you want to keep this?

File: packages/3dom/src/end-to-end-spec.ts [hunk 2]
```diff
@@ -71,4 +71,27 @@ suite('end-to-end', () => {
...
+    executionContext.changeModel(graft);
+
+    const messageEventArrives =
+        waitForEvent<MessageEvent>(executionContext.worker, 'message');
+    executionContext.eval('self.postMessage(model.materials[0].name)');
+
+    const messageEvent = await messageEventArrives;
+    console.log(messageEvent.data);
+
+    expect(messageEvent.data).to.be.equal(material.name);
```
Comment (elalish) [inline packages/3dom/src/end-to-end-spec.ts]: Does the astronaut have a material name, or are we just checking that empty string equals empty string?

File: packages/3dom/src/facade/three-js/three-dom-element.ts [hunk 3]
```diff
@@ -71,12 +71,17 @@ export class ThreeDOMElement implements ThreeDOMElementInterface {
...
-          null;
+
+    // NOTE: Some Three.js object names are modified from the names found in the
+    // glTF. Special casing is handled here, but might be better moved to
+    // subclasses down the road:
+    if ((relatedObject as Material).isMaterial) {
+      // Material names can be safely referenced directly from the Three.js
+      // object.
+      // @see: https://github.com/mrdoob/three.js/blob/790811db742ea9d7c54fe28f83865d7576f14134/examples/js/loaders/GLTFLoader.js#L2162
+      return (relatedObject as Material).name;
```
Comment (elalish) [inline packages/3dom/src/facade/three-js/three-dom-element.ts]: :+1:

File: packages/3dom/src/context.ts (hunk 1 above)
Comment (cdata) [reply packages/3dom/src/context.ts]: Thanks, will fix

File: packages/3dom/src/end-to-end-spec.ts (hunk 2 above)
Comment (cdata) [reply packages/3dom/src/end-to-end-spec.ts]: It has a name, I checked... `Astronaut_mat`, I think

Comment (elalish) [reply packages/3dom/src/end-to-end-spec.ts]: SG; you could add a check that the name is not empty here in the test as documentation.

**Pull request merged**